src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -1450,18 +1450,20 @@ public ResultSet getTables(
       // a schema if the current schema a user is connected to is different
       // given that we don't support show tables without a known schema.
       if (schemaPattern == null || isSchemaNameWildcardPattern(schemaPattern)) {
-        showCommand += " in database \"" + catalogEscaped + "\"";
+        showTablesCommand += " in database \"" + catalogEscaped + "\"";
       } else if (schemaPattern.isEmpty()) {
         return SnowflakeDatabaseMetaDataResultSet.getEmptyResultSet(GET_TABLES, statement);
       } else {
         String schemaUnescaped = unescapeChars(schemaPattern);
-        showCommand += " in schema \"" + catalogEscaped + "\".\"" + schemaUnescaped + "\"";
+        showTablesCommand += " in schema \"" + catalogEscaped + "\".\"" + schemaUnescaped + "\"";
       }
     }
 
-    logger.debug("sql command to get table metadata: {}", showCommand);
+    logger.debug("sql command to get table metadata: {}", showTablesCommand);
+    // for internal testing
+    showCommand = showTablesCommand;
```

Comment:
What's internal testing? Does it mean we don't have race condition test?

Reply:
The showCommand class variable was added previously for testing: https://github.com/snowflakedb/snowflake-jdbc/commit/387989cd6674485ab0c19d332e493803caf2c466

https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataInternalLatestIT.java#L54

Reply:
Can you remove showCommand variable from the code and in the test use reflection to get appropriate new variable like showColumnCommand or showTableCommand? It's not a good practice to expose methods just for testing. Also still do you able to repro race condition? If yes, can we add a new test case for it?

Reply:
I don't think we can use reflection to test the showColumnCommand or showTableCommand, these are local variables within the getColumns() and getTables() functions. I added a test case for the race condition.

Reply:
ok. Can you remove method getShowCommand() and in the test where ever it's use, get the "showCommand" value using reflection?

Reply:
Ideally test shouldn't check which command used. It should check the result value of getTables to confirm schema and table name. Can you change that way? Then we can remove showCommand variable all together.

Reply:
@sfc-gh-igarish updated to remove the showCommand and check the value of the result set to confirm schema.